config/ontoemma_nn_all.json

```diff
@@ -0,0 +1,107 @@
+{
+  "dataset_reader": {
+    "type": "ontology_matcher",
+    "name_token_indexers": {
+      "tokens": {
+        "type": "single_id",
+        "lowercase_tokens": true
+      },
+      "token_characters": {
+        "type": "characters"
+      }
+    },
+    "token_only_indexer": {
+      "tokens": {
+        "type": "single_id",
+        "lowercase_tokens": true
+      }
+    }
+  },
+  "train_data_path": "/net/nfs.corp/s2-research/scigraph/data/ontoemma/training/ontoemma.context.train",
+  "validation_data_path": "/net/nfs.corp/s2-research/scigraph/data/ontoemma/training/ontoemma.context.dev",
+  "model": {
+    "type": "ontoemmaNN",
+    "name_text_field_embedder": {
+            "tokens": {
+                    "type": "embedding",
+                    "pretrained_file": "/net/nfs.corp/s2-research/scigraph/data/ontoemma/embeddings/weights100.txt.gz",
+                    "embedding_dim": 100,
+                    "trainable": false
+            },
+            "token_characters": {
+                    "type": "character_encoding",
+                    "embedding": {
+                      "embedding_dim": 8
+                    },
+                    "encoder": {
+                      "type": "cnn",
+                      "embedding_dim": 8,
+                      "num_filters": 50,
+                      "ngram_filter_sizes": [5]
+                    },
+                    "dropout": 0.2
+            }
+    },
+    "context_text_field_embedder": {
+            "tokens": {
+                    "type": "embedding",
+                    "pretrained_file": "/net/nfs.corp/s2-research/scigraph/data/ontoemma/embeddings/weights100.txt.gz",
+                    "embedding_dim": 100,
+                    "trainable": false
+            }
+    },
+    "name_rnn_encoder": {
+      "type": "rnn",
+      "input_size": 150,
+      "hidden_size": 50,
+      "num_layers": 2,
+      "dropout": 0.2,
+      "bidirectional": true
+    },
+    "context_encoder": {
+      "type": "lstm",
+      "input_size": 100,
+      "hidden_size": 50,
+      "num_layers": 2,
+      "dropout": 0.2,
+      "bidirectional": true
+    },
+    "siamese_feedforward": {
+      "input_dim": 600,
+      "num_layers": 2,
+      "hidden_dims": 100,
+      "activations": "relu",
+      "dropout": 0.2
+    },
+    "decision_feedforward": {
+      "input_dim": 200,
+      "num_layers": 2,
+      "hidden_dims": [200, 1],
+      "activations": ["relu", "linear"],
+      "dropout": [0.2, 0.0]
+    }
+  },
```

Comment:
You might want to add an `initializer` field here. Here are a few suggestions:
- Use `xavier_normal` intialisation for all of your feedforward layers and the input -> hidden_state weights of the RNNs.
- Use `orthogonal` initialisation for the hidden_state -> hidden_state weights in the RNNs. These are called something very cryptic in pytorch, like `weight_hh`, so you might have to dig around a little to find out what the names are. If you look in the allennlp logs the parameter names should be logged.